extensions/mongodb/src/test/java/com/hazelcast/jet/mongodb/MongoDBSourceTest.java

```diff
@@ -16,339 +16,385 @@
+import static com.mongodb.client.model.Filters.gte;
+import static com.mongodb.client.model.Projections.include;
+import static com.mongodb.client.model.Sorts.ascending;
+import static java.util.Arrays.asList;
+import static java.util.Comparator.comparingInt;
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.IntStream.range;
 import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertNotNull;
 import static org.junit.Assert.assertNull;
 
+@RunWith(HazelcastParametrizedRunner.class)
+@Category({QuickTest.class})
 public class MongoDBSourceTest extends AbstractMongoDBTest {
 
+    private static final int COUNT_IN_BATCH = 10;
+    private static final int FILTERED_BOUND = 5;
+    @Parameter(0)
+    public boolean filter;
+    @Parameter(1)
+    public boolean projection;
+    @Parameter(2)
+    public boolean sort;
+    @Parameter(3)
+    public boolean map;
+
+    @Parameters(name = "filter:{0} | projection: {1} | sort: {2} | map: {3}")
+    public static Object[] filterProjectionSortMatrix() {
+        Set<Boolean> booleans = new HashSet<>(asList(true, false));
+        return Sets.cartesianProduct(booleans, booleans, booleans, booleans).stream()
+                   .map(tuple -> tuple.toArray(new Object[0]))
+                   .toArray(Object[]::new);
+    }
     @Test
-    public void testStream_whenServerDown() {
-        HazelcastInstance serverToShutdown = createHazelcastInstance();
-
-        int itemCount = 40_000;
-
-        Sink<Integer> setSink = SinkBuilder
-                .sinkBuilder("set", c -> c.hazelcastInstance().getSet("set"))
-                .<Integer>receiveFn(Set::add)
-                .build();
-
-
-        Pipeline p = Pipeline.create();
-        p.readFrom(streamSource(30))
-         .withNativeTimestamps(0)
-         .map(doc -> doc.getInteger("key"))
-         .writeTo(setSink);
-
-        JobConfig jobConfig = new JobConfig();
-        jobConfig.setProcessingGuarantee(ProcessingGuarantee.EXACTLY_ONCE)
-                 .setSnapshotIntervalMillis(2000);
-        Job job = hz.getJet().newJob(p, jobConfig);
-        ISet<Integer> set = hz.getSet("set");
-
-        spawn(() -> {
-            for (int i = 0; i < itemCount; i++) {
-                collection().insertOne(new Document("key", i));
-            }
-        });
-
-        sleepSeconds(5);
-
-        serverToShutdown.shutdown();
-
-        assertTrueEventually(() -> {
-            assertEquals(itemCount, set.size());
-        });
-
-        job.cancel();
-
+    public void testBatchSimple() {
+        IList<Object> list = instance().getList(testName.getMethodName());
+        final String connectionString = mongoContainer.getConnectionString();
+
+        collection().insertMany(range(0, COUNT_IN_BATCH)
+                .mapToObj(i -> newDocument("key", i).append("val", i))
+                .collect(toList())
+        );
+        assertEquals(collection().countDocuments(), COUNT_IN_BATCH);
```

Comment:
This (or similar snippet) is duplicated several times, you can extract it to some utility method.

Reply:
6f18195af4f9493afd5598b62b421744331792b8